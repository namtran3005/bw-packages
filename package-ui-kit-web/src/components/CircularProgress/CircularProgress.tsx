import * as React from 'react';
import styled from 'styled-components';
import { CircularProgress as MuiCircularProgress } from '@material-ui/core';
import { CircularProgressProps as MuiCircularProgressProps } from '@material-ui/core/CircularProgress';
import { ColorsType, Colors, Spacing } from '@bitwala-cryptobank-squad/package-theme';

type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;

interface CircularProgressProps
  extends Omit<MuiCircularProgressProps, 'color'> {
  foreground?: ColorsType;
  innerRef?: any; // eslint-disable-line @typescript-eslint/no-explicit-any
  thickness?: number;
}

const CircularProgressRootComponent: React.FunctionComponent<CircularProgressProps> = ({
  thickness = 4,
  ...props
}) => <MuiCircularProgress {...props} thickness={thickness} />;
const CircularProgressComponent = styled(CircularProgressRootComponent)`
  &&& {
    border: none;
    height: ${Spacing.unit * 2}px;
    color: ${({ foreground }) =>
      Colors.get(foreground !== undefined ? foreground : 'primaryBlack')};
  }
`;

const CircularProgress: React.FunctionComponent<CircularProgressProps> = (
  props
) => {
  return <CircularProgressComponent {...props} />;
};

export { CircularProgress, CircularProgressProps };
