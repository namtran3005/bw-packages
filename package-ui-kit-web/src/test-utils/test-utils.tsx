import {
  render as rtlRender,
  waitFor,
  RenderOptions as RTLRenderOptions,
} from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import { WebAppThemeProvider } from '../theme';
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';

type RenderOptions = RTLRenderOptions;

function renderWithTestHarness(
  ui: React.ReactElement,
  renderOptions: RenderOptions = {}
) {
  function Wrapper({ children }: any) {
    return (
      <MemoryRouter>
        <WebAppThemeProvider>{children}</WebAppThemeProvider>
      </MemoryRouter>
    );
  }

  return rtlRender(ui, {
    wrapper: Wrapper,
    ...renderOptions,
  });
}

// Use if you're in a hurry (better than nothing)
// be sure to verify no memory leak from implicit
// waitFor by adding these flags to jest: --runInBand --detectOpenHandles
async function renderAct(ui: React.ReactElement, options: RenderOptions = {}) {
  const view = renderWithTestHarness(ui, options);
  return await waitFor(() => view, {
    timeout: 5000,
    onTimeout: (e) => {
      // eslint-disable-next-line no-console
      console.error(e);
      return e;
    },
  });
}

export * from '@testing-library/react';

export { renderWithTestHarness, renderAct, userEvent };
