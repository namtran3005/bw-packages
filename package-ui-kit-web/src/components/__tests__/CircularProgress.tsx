import * as React from 'react';

import { renderWithTestHarness } from '../../test-utils/test-utils';
import { CircularProgress } from '../CircularProgress';

const defaultProps = {};

describe('CircularProgress', () => {
  it('should render circular progress', () => {
    const { baseElement } = renderWithTestHarness(
      <CircularProgress {...defaultProps} />
    );
    expect(baseElement).toMatchSnapshot();
  });
  it('should render determinate circular progress', () => {
    const { baseElement } = renderWithTestHarness(
      <CircularProgress variant="determinate" {...defaultProps} />
    );
    expect(baseElement).toMatchSnapshot();
  });
  it('should render indeterminate circular progress', () => {
    const { baseElement } = renderWithTestHarness(
      <CircularProgress variant="indeterminate" {...defaultProps} />
    );
    expect(baseElement).toMatchSnapshot();
  });
  it('should render circular progress custom foreground', () => {
    const { baseElement } = renderWithTestHarness(
      <CircularProgress
        variant="determinate"
        foreground="success"
        {...defaultProps}
      />
    );
    expect(baseElement).toMatchSnapshot();
  });
});
