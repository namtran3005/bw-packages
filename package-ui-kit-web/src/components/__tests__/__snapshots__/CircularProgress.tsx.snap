// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircularProgress should render circular progress 1`] = `
.c0.c0.c0 {
  border: none;
  height: 16px;
  color: #2C232E;
}

<body>
  <div>
    <div
      class="MuiCircularProgress-root c0 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="20"
          stroke-width="4"
        />
      </svg>
    </div>
  </div>
</body>
`;

exports[`CircularProgress should render circular progress custom foreground 1`] = `
<body>
  .c0.c0.c0 {
  border: none;
  height: 16px;
  color: #1CC18C;
}

<div>
    <div
      aria-valuenow="0"
      class="MuiCircularProgress-root c0 MuiCircularProgress-colorPrimary"
      foreground="success"
      role="progressbar"
      style="width: 40px; height: 40px; transform: rotate(0.000deg);"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="20"
          stroke-width="4"
          style="stroke-dasharray: 125.664; stroke-dashoffset: 125.664px;"
        />
      </svg>
    </div>
  </div>
</body>
`;

exports[`CircularProgress should render determinate circular progress 1`] = `
<body>
  .c0.c0.c0 {
  border: none;
  height: 16px;
  color: #2C232E;
}

<div>
    <div
      aria-valuenow="0"
      class="MuiCircularProgress-root c0 MuiCircularProgress-colorPrimary"
      role="progressbar"
      style="width: 40px; height: 40px; transform: rotate(0.000deg);"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="20"
          stroke-width="4"
          style="stroke-dasharray: 125.664; stroke-dashoffset: 125.664px;"
        />
      </svg>
    </div>
  </div>
</body>
`;

exports[`CircularProgress should render indeterminate circular progress 1`] = `
<body>
  .c0.c0.c0 {
  border: none;
  height: 16px;
  color: #2C232E;
}

<div>
    <div
      class="MuiCircularProgress-root c0 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="20"
          stroke-width="4"
        />
      </svg>
    </div>
  </div>
</body>
`;
